File: Python_Stack/Core_Assignments/ninjas_vs_pirates/classes/ninja.py
class Ninja:

    def __init__( self , name ):
        self.name = name
        self.strength = 10
        self.speed = 5
        self.health = 100
    
    def show_stats( self ):
        print(f"Name: {self.name}\nStrength: {self.strength}\nSpeed: {self.speed}\nHealth: {self.health}\n")

    def attack( self , pirate ):
        pirate.health -= self.strength
        print (f"{self.name} wacks pirate")
    
    def heal( self ):
        heal_val = (self.health/20) 
        self.health += heal_val
        print(f"{self.name} heals for{heal_val}")
    
    def crit( self , pirate ):
        pirate.health -= self.strength*2
        print (f"{self.name} stabs pirate")
    
    def ninja_star(self , pirate):
        pirate.health -= self.strength*50
        print(f"Ninja used shuriken")
        print("Instant K.O.")

File: MERN_Stack/Optional_Assignments/Ninja.js
// Assignment Ninja 

class Ninja{
    constructor(name, health = 1000, speed = 3, strength = 3){
        this.name = name;
        this.health = health;
        this.speed = speed;
        this.strength = strength;
    }

    sayName(){
        console.log(this.name);
            return this
    }

    showStats(){
        console.log(`Name: ${this.name} Strength: ${this.strength} Speed: ${this.speed} Health: ${this.health}`);
            return this
    }

    drinkSake(){
        this.health += 10;
        return this
    }
}

// Running Function 

const ninja = new Ninja("Hyabusa");

ninja.sayName().drinkSake().showStats();

